CONTRIBUTING.rst

```diff
@@ -42,23 +42,48 @@ Contributing
 
 **Step 1**: Create an `issue <https://github.com/rucio/rucio/issues/new>`_ with the description
 of the contribution (motivation, modification and expected results).
-Every issue will get a **unique issue number**.
+Every issue will get a **unique issue number**. Valid component names are
+listed in the `label list <https://github.com/rucio/rucio/labels>`_.
 
 **Step 2**: Create a local branch that corresponds to the issue. There are utility scripts to help you with this::
 
-  $ ./tools/create-patch-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
-  $ ./tools/create-feature-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
+  $ ./tools/create-patch-branch <unique issue number> '<short_change_message> #<issue number>: '
+  $ ./tools/create-feature-branch <unique issue number> '<short_change_message> #<issue number>'
+
+If you add a `github-recognised keyword <https://help.github.com/articles/closing-issues-using-keywords/>`_ then
+the associated issue can be closed automatically once the pull request is merged, e.g.::
+
+<short_change_message> Fix #<issue number>
+
+If you use different tools to create pull requests like the `github interface <https://help.github.com/articles/creating-a-pull-request/>`_
```

Comment:
This part should be in Step 4, not step 2 (creating the branch)

Reply:
I did a confusion.  I'm wondering if an option  --fixed should not be added to /tools/submit-pull-request to add the github keywords to the pull request title.

Reply:
For me personally, I just add the Keyword in the commit message, if the ticket really should be closed. I don't see a big advantage to have --fixed in the tool, as the commit message still has to be written by hand, but it's also fine for me to add it. It's just a search for the # and add a Fix before 😄

Reply:
it is not clear than the commit message is propagated in the pull request title in the text. added it.